source/validate_id.cpp

```diff
@@ -1216,22 +1216,21 @@ bool idUsage::isValid<SpvOpFunctionParameter>(const spv_instruction_t* inst,
                                    << inst->words[resultTypeIndex]
                                    << "' is not defined.";
            return false);
-  auto function = inst - 1;
   // NOTE: Find OpFunction & ensure OpFunctionParameter is not out of place.
   size_t paramIndex = 0;
-  while (firstInst != function) {
-    spvCheck(SpvOpFunction != function->opcode &&
-                 SpvOpFunctionParameter != function->opcode,
+  assert(firstInst < inst && "Invalid instruction pointer");
+  while (firstInst != --inst) {
```

Comment:
Ok.  I see that firstInst is initialized to the start of the copied instructions array.  So this firstInst != --inst actually will terminate, and won't result in an out-of-bounds access.